### PR TITLE
RPKI status field on hijack alerts

### LINKED ISSUE #252: Show RPKI status for routes seen in hijack alerts
**Is your feature request related to a problem? Please describe.**
New feature: when a hijack alert is detected, consult also RPKI ROAs related to the protected prefix (if they exist) to determine their status: valid, invalid, unknown.

**Describe the solution you'd like**
Additional field for rpki status for new hijack alerts.

**Additional context**
Check https://github.com/RIPE-NCC/rpki-validator-3 or rtrlib (preferrably we need sth with a Python interface).

### PR DESCRIPTION
**Pull request by vkotronis**
### Description of PR

What component(s) does this PR affect?

- [X] Back-End (Database, Detection/Configuration/etc. Microservices)
- [X] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs (incl. wiki)
- [ ] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Detection/Configuration/etc. Microservices)
- [ ] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [X] Docs (incl. wiki)
- [ ] Build System

### Related Issue

Resolves #252

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [X] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [X] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

Need to update wiki afterwards.

### CODE REVIEW

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **670**
- Hijacks Detected (hijacks/sec): **25**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **753**
- Hijacks Detected (hijacks/sec): **21**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **685**
- Hijacks Detected (hijacks/sec): **25**

**Comment by vkotronis** [comment]
TODO: RPKI Val container-microservice, enable it to answer to queries from detection (or get the info in-memory...)

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **705**
- Hijacks Detected (hijacks/sec): **22**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **744**
- Hijacks Detected (hijacks/sec): **23**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **659**
- Hijacks Detected (hijacks/sec): **26**

**Comment by vkotronis** [comment]
Need to solve error:
```
>>> from rtrlib import RTRManager, PfxvState
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/pyenv/versions/3.6.8/lib/python3.6/site-packages/rtrlib-0.1-py3.6-linux-x86_64.egg/rtrlib/__init__.py", line 12, in <module>
    from .rtr_manager import RTRManager, PfxvState
  File "/usr/local/lib/pyenv/versions/3.6.8/lib/python3.6/site-packages/rtrlib-0.1-py3.6-linux-x86_64.egg/rtrlib/rtr_manager.py", line 17, in <module>
    from _rtrlib import ffi, lib
ImportError: Error loading shared library librtr.so.0: No such file or directory (needed by /usr/local/lib/pyenv/versions/3.6.8/lib/python3.6/site-packages/rtrlib-0.1-py3.6-linux-x86_64.egg/_rtrlib.abi3.so)
>>> 
```

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **615**
- Hijacks Detected (hijacks/sec): **24**

**Comment by vkotronis** [comment]
https://github.com/rtrlib/python-binding/issues/12

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **661**
- Hijacks Detected (hijacks/sec): **23**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **647**
- Hijacks Detected (hijacks/sec): ****

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **681**
- Hijacks Detected (hijacks/sec): **26**

**Comment by vkotronis** [comment]
TODO: write RPKI val tests!

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **684**
- Hijacks Detected (hijacks/sec): **25**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **703**
- Hijacks Detected (hijacks/sec): **18**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **701**
- Hijacks Detected (hijacks/sec): **25**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **685**
- Hijacks Detected (hijacks/sec): **25**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **738**
- Hijacks Detected (hijacks/sec): **26**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **658**
- Hijacks Detected (hijacks/sec): **24**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **702**
- Hijacks Detected (hijacks/sec): **23**

**Comment by vkotronis** [comment]
TODO: show rpki status more prominently in the frontend, enable filtering with "LIKE"

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **764**
- Hijacks Detected (hijacks/sec): **26**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **702**
- Hijacks Detected (hijacks/sec): **26**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **653**
- Hijacks Detected (hijacks/sec): **27**

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **779**
- Hijacks Detected (hijacks/sec): **26**

**Comment by vkotronis** [comment]
Currently for testing:
```
  bgpstreamkafka:
    host: bmp.bgpstream.caida.org
    port: 9092
    topic: "^openbmp\\.router--.+\\.peer-as--.+\\.bmp_raw"
```

File: backend/Dockerfile [hunk 1]
```diff
@@ -2,15 +2,31 @@ FROM mavromat/alpine-python:3.6
...
 LABEL maintainer="Dimitrios Mavrommatis <jim.mavrommatis@gmail.com>"
 
-RUN apk update && apk add --no-cache supervisor tcl postgresql-client postgresql-dev redis curl
+RUN apk update && apk add --no-cache supervisor tcl postgresql-client postgresql-dev redis curl git libffi-dev cmake libssh
 
 WORKDIR /root
 
 COPY . ./
 
+RUN git clone https://github.com/rtrlib/rtrlib.git
```
Comment (slowr) [inline backend/Dockerfile]: this always downloads the latest, maybe version lock it?

File: backend/Dockerfile [hunk 2]
```diff
@@ -2,15 +2,31 @@ FROM mavromat/alpine-python:3.6
...
+WORKDIR /root/rtrlib
+RUN cmake -D CMAKE_BUILD_TYPE=Release .
+RUN make && make install
```
Comment (slowr) [inline backend/Dockerfile]: make -j 4 for some parallelization

File: backend/Dockerfile [hunk 3]
```diff
@@ -2,15 +2,31 @@ FROM mavromat/alpine-python:3.6
...
 RUN pip --no-cache-dir install -r requirements.txt \
-    && pip --no-cache-dir install git+https://github.com/supervisor/supervisor@a0ee8f1026c929ae4d9fc84741924414e8008f49 \
-    && make clean && make -j 4 \
+    && pip --no-cache-dir install git+https://github.com/supervisor/supervisor@a0ee8f1026c929ae4d9fc84741924414e8008f49
+
+RUN ln -s /usr/local/lib64/librtr.so /usr/local/lib/librtr.so \
+    && ln -s /usr/local/lib64/librtr.so.0 /usr/local/lib/librtr.so.0 \
+    && ln -s /usr/local/lib64/librtr.so.0.7.0 /usr/local/lib/librtr.so.0.7.0
+
+RUN git clone https://github.com/rtrlib/python-binding.git
```
Comment (slowr) [inline backend/Dockerfile]: same as above

File: backend/core/detection.py [hunk 4]
```diff
@@ -122,6 +127,29 @@ def __init__(self, connection: Connection) -> NoReturn:
             self.redis = redis.Redis(host=REDIS_HOST, port=REDIS_PORT)
             ping_redis(self.redis)
 
+            self.rtrmanager = None
+            if RPKI_VALIDATOR_ENABLED == "true":
```
Comment (slowr) [inline backend/core/detection.py]: if it's true false maybe they can be boolean already

File: backend/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,6 @@
+cffi>=1.4.0
 Cython==0.29.3
+enum34
```
Comment (slowr) [inline backend/requirements.txt]: versions for these packages?

File: backend/Dockerfile (hunk 1 above)
Comment (vkotronis) [reply backend/Dockerfile]: how?

File: backend/Dockerfile (hunk 3 above)
Comment (vkotronis) [reply backend/Dockerfile]: how can I lock this from source to a specific version?

File: backend/core/detection.py (hunk 4 above)
Comment (vkotronis) [reply backend/core/detection.py]: no they are strings as you can see in .env. It does not support booleans.

File: backend/requirements.txt (hunk 5 above)
Comment (vkotronis) [reply backend/requirements.txt]: I need to check them on the working test deployment and freeze them.

Comment (vkotronis) [reply backend/requirements.txt]: done. Resolving.

File: backend/Dockerfile (hunk 2 above)
Comment (vkotronis) [reply backend/Dockerfile]: only on make or also make install?

File: backend/Dockerfile (hunk 3 above)
Comment (vkotronis) [reply backend/Dockerfile]: done. Resolving

File: backend/Dockerfile (hunk 1 above)
Comment (vkotronis) [reply backend/Dockerfile]: done, resolving.

**Pull request merged**